controller/src/main/java/uni/miskolc/ips/ilona/tracking/controller/history/HistoryController.java

```diff
@@ -0,0 +1,109 @@
+package uni.miskolc.ips.ilona.tracking.controller.history;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.*;
+import uni.miskolc.ips.ilona.tracking.controller.model.UserPositionDTO;
+import uni.miskolc.ips.ilona.tracking.controller.util.UserPositionDTOConverter;
+import uni.miskolc.ips.ilona.tracking.model.UserPosition;
+import uni.miskolc.ips.ilona.tracking.service.HistoryService;
+import uni.miskolc.ips.ilona.tracking.service.exceptions.ServiceGeneralErrorException;
+import uni.miskolc.ips.ilona.tracking.service.exceptions.UserNotFoundException;
+
+import javax.xml.datatype.DatatypeConfigurationException;
+import java.sql.Timestamp;
+import java.util.List;
+
+@Controller
+@RequestMapping(value = "history")
+public class HistoryController {
+
+    private final Logger LOG = LogManager.getLogger(HistoryController.class);
+
+    private HistoryService historyService;
+
+    @Autowired
+    public HistoryController(HistoryService historyService) {
+        this.historyService = historyService;
+    }
+
+    @ResponseStatus(value = HttpStatus.CREATED)
+    @RequestMapping(value = "add", method = RequestMethod.GET, consumes = MediaType.APPLICATION_JSON_VALUE)
+    public void addHistory(@RequestBody UserPositionDTO userPositionDTO) throws ServiceGeneralErrorException, UserNotFoundException {
+        UserPosition userPosition = UserPositionDTOConverter.convertToUserPosition(userPositionDTO);
+        try {
+            historyService.addHistory(userPosition.getPosition(), userPosition.getTime(), userPosition.getUser().getUserid());
+        } catch (UserNotFoundException e) {
```

Comment:
You just rethrow the same exception after logging?

Reply:
Yes. In this way I can log the user ID and create an exception handler too